flow/raster_cache_unittests.cc

```diff
@@ -605,6 +605,59 @@ TEST(RasterCache, DisplayListWithSingularMatrixIsNotCached) {
   }
 }
 
+TEST(RasterCache, RasterCacheKeyHashFunction) {
+  RasterCacheKey::Map<int> map;
+  auto hash_function = map.hash_function();
+  SkMatrix matrix = SkMatrix::I();
+  uint64_t id = 5;
+  RasterCacheKey layer_key(id, RasterCacheKeyKind::kLayer, matrix);
+  RasterCacheKey picture_key(id, RasterCacheKeyKind::kPicture, matrix);
+  RasterCacheKey display_list_key(id, RasterCacheKeyKind::kDisplayList, matrix);
+
+  auto layer_hash_code = hash_function(layer_key);
+  ASSERT_EQ(layer_hash_code, fml::HashCombine(id, RasterCacheKeyKind::kLayer));
+
+  auto picture_hash_code = hash_function(picture_key);
+  ASSERT_EQ(picture_hash_code,
+            fml::HashCombine(id, RasterCacheKeyKind::kPicture));
+
+  auto display_list_hash_code = hash_function(display_list_key);
+  ASSERT_EQ(display_list_hash_code,
+            fml::HashCombine(id, RasterCacheKeyKind::kDisplayList));
+}
+
+TEST(RasterCache, RasterCacheKeySameID) {
+  RasterCacheKey::Map<int> map;
+  SkMatrix matrix = SkMatrix::I();
+  uint64_t id = 5;
+  RasterCacheKey layer_key(id, RasterCacheKeyKind::kLayer, matrix);
+  RasterCacheKey picture_key(id, RasterCacheKeyKind::kPicture, matrix);
+  RasterCacheKey display_list_key(id, RasterCacheKeyKind::kDisplayList, matrix);
+  map[layer_key] = 100;
+  map[picture_key] = 200;
+  map[display_list_key] = 300;
+
+  ASSERT_EQ(map[layer_key], 100);
+  ASSERT_EQ(map[picture_key], 200);
+  ASSERT_EQ(map[display_list_key], 300);
+}
+
+TEST(RasterCache, RasterCacheKeySameKind) {
+  RasterCacheKey::Map<int> map;
+  RasterCacheKeyKind kind = RasterCacheKeyKind::kDisplayList;
```

Comment:
I would repeat this test for kLayer and kPicture.

Reply:
Done